dell/xps/13-9310/default.nix

```diff
@@ -4,26 +4,13 @@
     ../../../common/pc/laptop
     ../../../common/pc/ssd
   ];
-  
+
 
   # Includes the Wi-Fi and Bluetooth firmware for the QCA6390.
   hardware.enableRedistributableFirmware = true;
 
+  # Requires at least 5.12 for working wi-fi and bluetooth.
   boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.12") (lib.mkDefault pkgs.linuxPackages_latest);
-  # TODO: upstream this to NixOS
-  boot.kernelPatches = [{
-    name = "enable-qca6390-bluetooth";
-    patch = null;
-    extraConfig = ''
-      BT_QCA m
-      BT_HCIUART m
-      BT_HCIUART_QCA y
-      BT_HCIUART_SERDEV y
-      SERIAL_DEV_BUS y
-      SERIAL_DEV_CTRL_TTYPORT y
-    '';
-  }];
-  boot.kernelModules = [ "btqca" "hci_qca" "hci_uart" ];
```

Comment:
Given that the next nixos release is soon, we can wait a bit otherwise, we can also set these patches already conditionally based on the nixpkgs version.

Reply:
Okydoke 21.11 has landed, I think this should be OK to land now